### ISSUE TITLE
Notifications with customizable sounds

### ISSUE THREAD

**Issue by iambillzhao** [p0deje/Maccy#211]
The current soundtracks for copy and paste are not distinct enough and seem to have significantly lower volume than all other notifications. It would be great to have users set their preferred soundtracks through _preferences -> advanced_.

As changing `write.mp3` and `knock.mp3` under package resources does not work as expected, it would be easier if they are replaceable from the preferences page.

**Comment by p0deje** [comment]
1. Would increase the volume of the sounds help?
2. Not sure how to make sounds customizable, but I'll merge PRs for that.

**Comment by peelman** [comment]
@p0deje so I came here to ask something else, but perhaps we can kill two birds here; a better strategy would probably be to use Notification Center.  What I came to ask for is the ability to show a notification, rather than just play a sound; but you could basically replace your call to NSSound with a [few lines to create a UserNotification](https://developer.apple.com/documentation/usernotifications) and you can [create a library of sounds within the app bundle](https://developer.apple.com/documentation/usernotifications/unnotificationsound) to call at will when sending the notification.

Would make Maccy a better Mac citizen too, because the notifications would be subject to DND, and you get a lot of free configuration by just letting users manage how the notification is displayed/managed via the global OS settings.  I _think_ there is also a way now to include that settings pane in your app (i have seen it done on iOS, though specific examples are escaping me; but i haven't seen it done on macOS).  I would really like to see this done and would be willing to work on the code to make it happen; but as it stands I can't even get `master` to build due to a `dyld` error telling me Sparkle isn't loaded.

Been a long while since i've done macOS dev, and I don't have a lot of time to devote to all the minutia right now.

**Comment by peelman** [comment]
For reference, I was able to get this working...here is the code i have that is _functional_ but I'm not proposing that it's complete:
```  
func add(_ item: HistoryItem) {
    if let existingHistoryItem = findSimilarItem(item) {
      item.contents = existingHistoryItem.contents
      item.firstCopiedAt = existingHistoryItem.firstCopiedAt
      item.numberOfCopies += existingHistoryItem.numberOfCopies
      item.pin = existingHistoryItem.pin
      remove(existingHistoryItem)
    } else {
      if UserDefaults.standard.playSounds {
        NSSound(named: NSSound.Name("write"))?.play()
        let notification = NSUserNotification()
          notification.title = "Clipboard Contents Changed"
          notification.informativeText = nil
          notification.soundName = NSUserNotificationDefaultSoundName
          NSUserNotificationCenter.default.deliver(notification)
      }
    }

    CoreDataManager.shared.saveContext()
  }
```

####  Sidebar:
This has already let me confirm that Firefox is doing _something_ that screws with Copying text fluidly and consistently.  Its like a 30% failure rate for copying text where it never actually makes it to the clipboard.  At first I thought it was just a delay; I was switching away from Firefox before it had a chance to complete the operation or something.  But nope; a significant amount of time, it just doesn't do the copy operation as requested.  Hammering on CMD-C 5 or 6 times, or a dozen, and it'll eventually go. 😡  At least _now_, I can have an audible (and visual) confirmation that my operation succeeded, but I do dumb things like paste passwords into Slack because I thought I had copied some text or a URL but Firefox decided "nope, not today!"

**Comment by p0deje** [comment]
I've spent a few hours today trying to figure out how to plug https://developer.apple.com/documentation/usernotifications into Maccy and I have to say that it's extremely confusing. My idea was to allow Maccy to send notifications on new copy/paste with custom sound, but:

1. I couldn't make custom sounds work at all - they simply always play the default alert sound.
2. There seems to be no way to allow to opt-in for alerts - only opt-out. This means that as long as Maccy has permission to send notifications - it will be sending them as alerts + sounds.
3. At first, there was a checkbox to disable sounds for Maccy in the "Notifications & Focus" preference, but now there is none. I can't understand why in the world would that happen.
4. There is no way to remove the app from the "Notifications & Focus" preference and I have weird bugs testing this all together.

I've drafted https://github.com/p0deje/Maccy/pull/411 to demo how it works, but at this point I see no value in continuing the struggle. Would be happy to see if anyone else chimes in and shares their perspective.
